test/IntegrationTest.php

```diff
@@ -75,6 +81,99 @@ public function testResolveIPv6only()
         });
     }
 
+    public function testResolveUsingSearchList()
+    {
+        Loop::run(function () {
+            $configLoader = \stripos(PHP_OS, "win") === 0
+                ? new WindowsConfigLoader()
+                : new UnixConfigLoader();
+            /** @var Dns\Config $config */
+            $config = yield $configLoader->loadConfig();
+            /** @var Dns\ConfigLoader|MockObject $configLoader */
+            $configLoader = $this->createMock(Dns\ConfigLoader::class);
+            $configLoader->expects($this->once())
+                ->method('loadConfig')
+                ->willReturn(new Success(new Dns\Config(
+                    $config->getNameservers(),
+                    $config->getKnownHosts(),
+                    $config->getTimeout(),
+                    $config->getAttempts(),
+                    ['kelunik.com'],
+                    1
+                )));
+
+            Dns\resolver(new Dns\Rfc1035StubResolver(null, $configLoader));
+            $result = yield Dns\resolve('blog');
+
+            /** @var Record $record */
+            $record = $result[0];
+            $inAddr = @\inet_pton($record->getValue());
+            $this->assertNotFalse(
+                $inAddr,
+                "Server name blog.kelunik.com did not resolve to a valid IP address"
+            );
+        });
```

Comment:
Should probably assert that the result is the same as a query for `blog.kelunik.com`?